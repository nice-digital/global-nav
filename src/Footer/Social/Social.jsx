import FacebookIcon from "@nice-digital/icons/lib/Facebook";
import XLogo from "@nice-digital/icons/lib/XLogo";
import YouTubeIcon from "@nice-digital/icons/lib/YoutubePlay";
import LinkedInIcon from "@nice-digital/icons/lib/Linkedin";

import styles from "./Social.module.scss";
import TrackedLink from "../../TrackedLink";
import { footerClickEventAction } from "../../tracker";

const Social = () => {
	return (
		<div className={styles.wrapper}>
			<ul className={styles.list} aria-label="NICE on social media">
				<li>
					<TrackedLink
						href="https://en-gb.facebook.com/NationalInstituteforHealthandCareExcellence/"
						eventAction={footerClickEventAction}
						eventLabel="Facebook"
					>
						<FacebookIcon />
						Facebook
					</TrackedLink>
				</li>
				<li>
					<TrackedLink
						href="https://x.com/NICEcomms"
						eventAction={footerClickEventAction}
						eventLabel="X"
					>
						<XLogo />X
					</TrackedLink>
				</li>
				<li>
					<TrackedLink
						href="https://www.youtube.com/user/NICEmedia"
						eventAction={footerClickEventAction}
						eventLabel="YouTube"
					>
						<YouTubeIcon />
						YouTube
					</TrackedLink>
				</li>
				<li>
					<TrackedLink
						href="https://www.linkedin.com/company/national-institute-for-health-and-clinical-excellence"
						eventAction={footerClickEventAction}
						eventLabel="LinkedIn"
					>
						<LinkedInIcon />
						LinkedIn
					</TrackedLink>
				</li>
			</ul>
		</div>
	);
};

export default Social;
