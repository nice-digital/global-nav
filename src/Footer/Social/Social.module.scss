@use '@nice-digital/nds-core/scss/media-queries';
@use '@nice-digital/nds-core/scss/spacing';
@use '@nice-digital/nds-core/scss/utils';

.wrapper {
  clear: left;
}

.list {
  list-style: none;
  margin: 0;
  max-width: 100%;
  padding: 0;

  li {
    margin: utils.rem(0 0 spacing.$x-small);
  }

  svg {
    margin: utils.rem(-(spacing.$xx-small) spacing.$x-small 0 0);
    vertical-align: middle;
  }
}

@include media-queries.mq($from: sm) {
  .list {
    li {
      float: left;
      margin: utils.rem(0 spacing.$medium 0 0);
    }
  }
}

@include media-queries.mq($from: md) {
  .wrapper {
    clear: none;

    li {
      float: none;
      margin: utils.rem(0 0 spacing.$x-small);
    }
  }
}
