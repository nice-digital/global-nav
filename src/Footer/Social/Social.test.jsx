import React from "react";

import Social from "./Social";
import { shallow } from "enzyme";
import toJson from "enzyme-to-json";

describe("Social", () => {
	it("Renders without crashing", () => {
		const wrapper = shallow(<Social />);
		expect(wrapper).toHaveLength(1);
	});

	it("Matches snapshot", () => {
		const wrapper = shallow(<Social />);
		expect(toJson(wrapper)).toMatchSnapshot();
	});
});
