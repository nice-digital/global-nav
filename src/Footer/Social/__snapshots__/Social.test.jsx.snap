// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Social Matches snapshot 1`] = `
<div
  className="wrapper"
>
  <ul
    aria-label="NICE on social media"
    className="list"
  >
    <li>
      <TrackedLink
        eventAction="Footer click"
        eventCategory="TopHat and footer"
        eventLabel="Facebook"
        href="https://en-gb.facebook.com/NationalInstituteforHealthandCareExcellence/"
      >
        <SvgFacebook />
        Facebook
      </TrackedLink>
    </li>
    <li>
      <TrackedLink
        eventAction="Footer click"
        eventCategory="TopHat and footer"
        eventLabel="Twitter"
        href="https://twitter.com/NICEcomms"
      >
        <SvgTwitter />
        Twitter
      </TrackedLink>
    </li>
    <li>
      <TrackedLink
        eventAction="Footer click"
        eventCategory="TopHat and footer"
        eventLabel="YouTube"
        href="https://www.youtube.com/user/NICEmedia"
      >
        <SvgYoutubePlay />
        YouTube
      </TrackedLink>
    </li>
    <li>
      <TrackedLink
        eventAction="Footer click"
        eventCategory="TopHat and footer"
        eventLabel="Instagram"
        href="https://www.instagram.com/nicecomms/"
      >
        <SvgInstagram />
        Instagram
      </TrackedLink>
    </li>
    <li>
      <TrackedLink
        eventAction="Footer click"
        eventCategory="TopHat and footer"
        eventLabel="LinkedIn"
        href="https://www.linkedin.com/company/national-institute-for-health-and-clinical-excellence"
      >
        <SvgLinkedin />
        LinkedIn
      </TrackedLink>
    </li>
  </ul>
</div>
`;
